text/0000-channel-participation-api-without-system-channel.md

```diff
@@ -0,0 +1,536 @@
+- These OSNs should be configured to accept channel participation commands from their respective local orderer channel 
+  participation admins. This is achieved by setting the respective fields in the orderer.yaml file of each OSN.
+- These OSNs may belong to different organizations.
+- Generate the genesis block of a system channel. (The system channel contains the `Consortiums` config group, whereas 
+  application channels do not).
+- Create the system channel by invoking the "Join" command as described above, but supply the genesis block of the 
+  system channel, rather than that of an application channel.
+- The channel participation command "Join" will no longer accept commands, and channel creation can only 
+  be done using a config transaction on the system channel.
+- The channel participation commands "List" and "Delete" (of system channel only) will continue to function.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The implementation of the channel participation API can be split in two. One part deals with the "API layer" and 
+deals with exposing the API via an HTTP server. The second part receives abstract channel participation commands 
+via a Go interface, and executed the operations described above.  
+  
+## The API layer
+The peer and the orderer host an HTTP server that offers a RESTful “operations” API (see: 
+[The Operations Service](https://hyperledger-fabric.readthedocs.io/en/release-2.0/operations_service.html)).
+We intend to hook into the HTTP server that is hosted by the orderer, and expose a new capability 
+`/channel-participation`, which has a single resource  `/channels` as described above. 
+The channel participation commands will only be enabled if specified explicitly in the `orderer.yaml` 
+configuration file. These commands will only be available in the orderer and not in the peer.
+
+The _local orderer participation admin_  role will have the same authentication and authorization
+as the _operations admin_ (i.e. the same mutual TLS certificate). We choose this option because it is the simplest 
+to implement at this time. This is expected to change in the future (see below).
+
+## Participation management
+We intend to implement the channel participation management functionality in a new package 
+(`orderer/common/channelparticipation`). In this package a new class will keep a map of channel names and their status,
+and will be in charge of coordinating the operations of following or joining a channel as a member. 
+Every channel that needs to follow or catch-up on the cluster will have a dedicated goroutine. When the channel reaches
+a state where it can join the cluster, the respective methods in `orderer/common/multichannel/registrar.go` will be 
+invoked in order to start the chain runtime.
+
+When a channel is first joined with a config block, the channel's ledger folder is created, and inside it the config 
```

Comment:
This seems like it might be a bit racy -- we could crash after creating the ledger folder but before writing the block.  Perhaps it would be better to reverse the order and locate the block outside the ledger dir so that we could resume creating the ledger dir if interrupted?

Reply:
You are correct, this needs to be done atomically. One way to do this is to create a temporary folder with all the resources and then rename it into place. Renaming is atomic so if we fail before the rename nothing happens.